### DISCUSSION TITLE
Cannot remove margin between very top and navbar

### DISCUSSION THREAD

**Discussion by yvesbou** [heroui-inc/heroui#817] [Help] [answered]
Hey guys, I'm a beginner in frontend development. I want that my navbar has 0 margin to the top, but there is some. Also it doesn't span the full width. Even though I set width to 100 in both elements.

In my wrapping container I have margin 0, as well as in the navbar. How do I control this distance between the top and the navbar?
```
<div>
   <Container
        fluid
        as="main"
        display="flex"
        direction="column"
        justify="flex-start"
        alignItems="center"
        style={{ height: '100vh', margin: 0, width: '100vw'}}
        css={{background: '$gradient'}}
   >
      <Navbar isCompact isBordered variant="static" style={{boxSizing: "border-box", margin: 0, maxWidth: '100%', color: theme?.colors.secondary.value,
          fontSize: theme?.fontSizes.md.value, fontWeight: theme?.fontWeights.hairline.value,
          padding: `${theme?.space['xs'].value} ${theme?.space[4].value}`}}>
        <Navbar.Brand>
          <Text b color="inherit" hideIn="xs" css={{font: theme?.fonts.sans, fontSize: theme?.fontSizes['4xl'].value, fontWeight: theme?.fontWeights.black.value}}>
            CryptoDevs
          </Text>
        </Navbar.Brand>
        <Navbar.Content hideIn="xs" variant="highlight-rounded">
          <Navbar.Link isActive href="#">Whitelist</Navbar.Link>
          <Navbar.Link href="#">NFT Drop</Navbar.Link>
          <Navbar.Link href="#">Smart Contracts</Navbar.Link>
        </Navbar.Content>
        <Navbar.Content>
          <Navbar.Item>
            <ConnectButton/>
          </Navbar.Item>
        </Navbar.Content>
      </Navbar>
</div>
```

<img width="1274" alt="Screenshot 2022-10-12 at 05 46 31" src="https://user-images.githubusercontent.com/63062257/195245287-737d1252-7c37-44b0-9020-7c5c8bd2889a.png">

**Comment by tianenpang** [comment, marked as answer]
Hi @yvesbou please check the `maxWidth` prop in the `Navbar` [docs](https://nextui.org/docs/components/navbar#apis:~:text=maxWidth), when setting `maxWidth` to `fluid` the navbar will fill up the width of the screen.

For the gap at the top, check if it's caused by the parent element 🤔

Here's an example: [codesandbox.io/s/nextui-discussion-817-fmp0sk](https://codesandbox.io/s/nextui-discussion-817-fmp0sk)

**Reply by Xariwey** [reply]
fluid prop in container is broken thats why he is manually setting the maxwidth...
heres the tread #754

@yvesbou make sure ur wrapping ur app with nextuiprovider
https://nextui.org/docs/guide/getting-started#setup

**Reply by tianenpang** [reply]
Yeah, there seems to be a conflict with `responsive` 🤔 so I set [`responsive={false}`](https://codesandbox.io/s/nextui-discussion-817-fmp0sk?file=%2Fsrc%2FApp.js%3A1477-1523).

**Reply by yvesbou** [reply]
I have wrapped the app with nextuiprovider.

```jsx
<WagmiConfig client={wagmiClient}>
      <RainbowKitProvider chains={chains}>
        <NextUIProvider>
           <Component {...pageProps} />
        </NextUIProvider>
      </RainbowKitProvider>
</WagmiConfig>
```

**Reply by yvesbou** [reply]
The reason why there was a space to the top was the padding css property 

`padding: ``${theme?.space['xs'].value} ${theme?.space[4].value}`

However,

```jsx
<Navbar isCompact isBordered variant="static" style={{boxSizing: "border-box", margin: 0, maxWidth: 'fluid', color: theme?.colors.secondary.value,
          fontSize: theme?.fontSizes.md.value, fontWeight: theme?.fontWeights.hairline.value,
          }}>
```

doesn't fill the full width even with `maxWidth: 'fluid'`

**Reply by tianenpang** [reply]
The `maxWidth` is a variant of [Navbar](https://nextui.org/docs/components/navbar#navbar-props:~:text=maxWidth), setting `maxWidth: 'fluid'` in `style` won't work.

```tsx
<Navbar
  isCompact
  isBordered
  variant="static"
  maxWidth="fluid" // add here
  style={{
    boxSizing: "border-box",
    margin: 0,
    // maxWidth: 'fluid', // remove
    color: theme?.colors.secondary.value,
    fontSize: theme?.fontSizes.md.value,
    fontWeight: theme?.fontWeights.hairline.value
  }}
>
```

**Reply by yvesbou** [reply]
well thanks for your quick answer and help, but I cannot make this work better

this is everything I have (I think the content inside _app.tsx and _document.tsx doesn't matter right?) and still small space to the left and to the right remain

```jsx
<div>
      <Head>
        <title>NextUI | Create Next App</title>
        <meta
          name="description"
          content="Generated by create next app and using NextUI as a react UI library"
        />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Container
        fluid
        responsive={false}
        as="main"
        display="flex"
        direction="column"
        justify="flex-start"
        alignItems="center"
        style={{ height: '100vh', margin: 0, width: '100vw'}}
        css={{background: '#16181A'}}
      >
      
        
          <Navbar isCompact disableBlur maxWidth="fluid" variant="static" style={{boxSizing: "border-box", margin: 0, color: 'white',
              fontSize: theme?.fontSizes.md.value, fontWeight: theme?.fontWeights.hairline.value
              }} containerCss={{backgroundColor: 'black'}}>
            <Navbar.Brand>
              <Text b color="inherit" hideIn="xs" css={{font: theme?.fonts.sans, fontSize: theme?.fontSizes['4xl'].value, fontWeight: theme?.fontWeights.black.value}}>
                CryptoDevs
              </Text>
            </Navbar.Brand>
            <Navbar.Content hideIn="xs" variant="highlight-rounded">
              <Navbar.Link isActive href="#">Whitelist</Navbar.Link>
              <Navbar.Link href="#">NFT Drop</Navbar.Link>
              <Navbar.Link href="#">Smart Contracts</Navbar.Link>
            </Navbar.Content>
            <Navbar.Content>
              <Navbar.Item>
                <ConnectButton/>
              </Navbar.Item>
            </Navbar.Content>
          </Navbar>
        
      </Container>
    </div>

```

**Reply by yvesbou** [reply]
do you have a repo @tianenpang where I can see a customised functioning navbar?

**Reply by tianenpang** [reply]
The `Container` has some `padding`, you can set in the `css={{ background: '#16181A', padding: 0 }}`.

By the way, I might not explained well [above](https://github.com/nextui-org/nextui/discussions/817?notification_referrer_id=NT_kwDOAfQQr7M0NTg3NDg1NTA1OjMyNzcyMjcx#discussioncomment-3866030), the reason why it doesn't work in `style` is because written in style it's just the regular css, but when passed in as a props, it'll be the variant of Navbar in order to apply the `fluid` styles.

**Reply by yvesbou** [reply]
oh yeah the padding in the container did it! thanks a lot 👍
